docs/guides/integration-javascript.md

```diff
@@ -0,0 +1,293 @@
+- [Configure for Hashed Email Address](#configure-for-hashed-email-address)
+- [Configure for Phone Number](#configure-for-phone-number)
+- [Configure for Hashed Phone Number](#configure-for-hashed-phone-number)
+
+If the SDK is configured multiples times, it uses the most recent configuration values.
+
+
+### Configure for Email Address
+
+Configure the UID2 SDK with an email address:
+
+```js
+await __uid2.setIdentityFromEmail(
+    "test@example.com",
+    {
+        subscriptionId: subscriptionId,
+        serverPublicKey: publicKey,
+    }
+);
+```
+
+No normalization or hashing is required by the publisher.
+
+The UID2 SDK normalizes and hashes the email address before sending the encrypted hash to the UID2 service.
+
+### Configure for Hashed Email Address
+
+Configure the UID2 SDK with a hashed email address:
+
+```js
+await __uid2.setIdentityFromEmailHash(
+    'eVvLS/Vg+YZ6+z3i0NOpSXYyQAfEXqCZ7BTpAjFUBUc=',
+    {
+        subscriptionId: subscriptionId,
+        serverPublicKey: publicKey,
+    }
+);
+```
+
+**The publisher is responsible for normalizing and hashing the email address**. For details, see [Normalization and Encoding](../getting-started/gs-normalization-encoding.md).
+
+The UID2 SDK encrypts the hash before sending it to the UID2 service.
+
+### Configure for Phone Number
+
+Configure the UID2 SDK with a phone number:
+
+```js
+await __uid2.setIdentityFromPhone(
+    '+1111111111',
+    {
+        subscriptionId: subscriptionId,
+        serverPublicKey: publicKey,
+    }
+);
+```
+
+**The publisher is responsible for normalizing the phone number**. For details, see [Phone Number Normalization](../getting-started/gs-normalization-encoding.md#phone-number-normalization).
+
+The UID2 SDK hashes the phone number before sending the encrypted hash to the UID2 service.
+
+### Configure for Hashed Phone Number
+
+Configure the UID2 SDK with a hashed phone number:
+
+```js
+await __uid2.setIdentityFromPhoneHash(
+    'eVvLS/Vg+YZ6+z3i0NOpSXYyQAfEXqCZ7BTpAjFUBUc=',
+    {
+        subscriptionId: subscriptionId,
+        serverPublicKey: publicKey,
+    }
+);
+```
+
+**The publisher is responsible for normalizing and hashing the phone number**. For details, see [Normalization and Encoding](../getting-started/gs-normalization-encoding.md).
+
+The UID2 SDK encrypts the hash before sending it to the UID2 service.
+
+## Token Storage and Refresh
+
+The generated UID2 token will be stored in local storage, under the key UID2-sdk-identity. The SDK will refresh the UID2 token periodically.
+
+## Retrieving the UID2 Advertising Token 
+
+With the setup above, when the UID2 SDK generates a UID2 token, the advertising token (TODO add link to glossary) will be ready to retrieve by calling the ```getAdvertisingToken```. The advertising token is the token publishers will pass to the bidstream. Before calling this function, be sure to call init() and wait until your callback handler has received an InitCompleted event.
```

Comment:
It's probably better to use the events mentioned above to receive the token - if they rely on `getAdvertisingToken` they won't receive changes to it and could end up with an old token even though a refreshed one becomes available straight after.

Reply:
changed to mention waiting for the IdentityUpdated event